server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -862,10 +868,10 @@ public void deleteStaleSegmentsAsync(int lastNMetadataFilesToKeep) {
      *
      * @param lastNMetadataFilesToKeep number of metadata files to keep
      */
-    public void deleteStaleSegmentsAsync(int lastNMetadataFilesToKeep, ActionListener<Void> listener) {
+    public void deleteStaleSegmentsAsync(int lastNMetadataFilesToKeep, ActionListener<Void> listener, String threadPoolName) {
```

Comment:
why do we need to take `threadpoolName` as an argument here as it should always be `ThreadPool.Names.REMOTE_PURGE` .

Reply:
was using it before to use `Threadpool.Names.SAME` to run the cleanup in the same thread. rolled back this change now as we don't need it anymore.